### PR TITLE
Beach Expansion Update

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Expands the beach away mission to include a brand new scuba-diving area!
- Somewhat empty, but has a few points of interest scattered about and can be added to in the future. Better mappers and spriters than myself would probably be needed to really make it look nice.

Visually updates the beach
- Sand texture changed, and given randomization for a more varied beachfront.
- The row of crates and lockers has been replaced with 3 changing rooms/tents so you can change into that mankini in private.
- Changes the water texture to be less bland
- Expands the water area to give more room for swimmers to play in before they reach the edge

Adds in the Diving Point Buoy and Diving Point Anchor objects, used to transition between the beach surface and the underwater area
- Hit with an empty hand to transition between the two points

The underwater area has a few "minor" dangers to prepare for:
- Sea Carp
 - These carp (unlike their space-faring cousins) won't attack on sight, and will try to swim away when attacked. They still bite if they work up the courage to come back.
- Water
 - Water and lungs don't work well together, who knew!? Skrell, Neara, and anyone who has NO_BREATHE can happily explore, while the rest will want to pack some internals lest they begin to drown.
 - Drowning in the scuba area doesn't require you to be lying down, unlike the pool. You're in over your head (literally).
- Darkness
 - Not quite a danger in itself, but it's easy to get disoriented in the dark water, and that can be frustrating.

CODE STUFF
- Broke up the pool controller code into a couple procs for cleanliness
and in case we want to have a type of pool controller that doesn't drown
or doesn't adjust bodytemp
- Made ladders use forceMove instead of directly setting loc
- Removed an entirely unused area (/area/beach)

Crappy coder-sprites for Buoy and Anchor done by myself.
- Ok, they actually aren't THAT bad, but they aren't fantastic.

NOT MAP-MERGED BECAUSE I'M A REBEL.
(And because map-merge supposedly breaks when changing map dimensions)

### CODE REVIEW

**Comment by Spacemanspark** [comment]
Can we get some pictures?
:+1:

File: code/game/machinery/poolcontroller.dm [hunk 1]
```diff
@@ -39,63 +42,73 @@
...
-					else
-						drownee.adjustOxyLoss(5) //5 oxyloss per cycle.
-						add_logs(drownee, src, "drowned")
-						if(prob(35)) //35% chance to tell them what is going on. They should probably figure it out before then.
-							drownee.visible_message("<span class='danger'>\The [drownee] flails, almost like they are drowning!</span>","<span class='userdanger'>You're lacking air!</span>") //*gasp* *gasp* *gasp* *gasp* *gasp*
-
-		for(var/obj/effect/decal/cleanable/decal in W)
+				handleDrowning(drownee)		//Only human types will drown, to keep things simple for non-human mobs that live in the water
+
+		for(var/obj/effect/decal/cleanable/decal in T)
```
Comment (tigercat2000) [inline code/game/machinery/poolcontroller.dm]: add comments explaining what each thing does please


no it's not really necessary but I spent about an hour making the thing as easy to understand for someone new as possible so do it

File: code/game/objects/structures/ladders.dm [hunk 2]
```diff
@@ -24,6 +29,9 @@
 		update_icon()
 
 /obj/structure/ladder/update_icon()
+	if(istype(src, /obj/structure/ladder/dive_point))
```
Comment (tigercat2000) [inline code/game/objects/structures/ladders.dm]: .... uhh

File: code/game/machinery/poolcontroller.dm [hunk 3]
```diff
@@ -39,63 +42,73 @@
...
-					if(prob(50)) //inform the mob of cold water half the time
-						M << "<span class='warning'>The water is chilly.</span>" //Inform the mob it's chilly water.
-
-				if("frigid") //Freeze the mob.
-					M.bodytemperature = max(80, M.bodytemperature - 35) //cool mob at -35k per cycle
-					M << "<span class='danger'>The water is freezing!</span>"
-
+	for(var/turf/T in linkedturfs) //Check for pool-turfs linked to the controller.
+		for(var/mob/M in T) //Check for mobs in the linked pool-turfs.
+			if(temperature != "normal")		//don't bother with this if we aren't going to be changing anything anyways
```
Comment (Tastyfish) [inline code/game/machinery/poolcontroller.dm]: Um, pretty sure you can change the temperature on it. Don't forget this is still a thing on the station.

**Comment by FalseIncarnate** [comment]
## Some pictures, but not too many so I don't ruin the fun of exploring:

image
A sunken home? Or perhaps it belonged to a Skrell?

image
Your beachfront may vary. (The water overlay still sucks and is blocky, but that's for another time)

File: code/game/machinery/poolcontroller.dm (hunk 3 above)
Comment (Tastyfish) [inline code/game/machinery/poolcontroller.dm]: Why don't you just handle "normal" in `handleTemp()`?

**Comment by Spacemanspark** [comment]
Perfect, thanks!

File: code/game/objects/structures/ladders.dm (hunk 2 above)
Comment (Tastyfish) [inline code/game/objects/structures/ladders.dm]: also yeah, wut. Use OOP

File: code/game/machinery/poolcontroller.dm (hunk 3 above)
Comment (FalseIncarnate) [inline code/game/machinery/poolcontroller.dm]: No clue, why it's not in there, but "normal" does nothing so I figured there was no point in even calling the proc if we won't be using it.

File: code/game/machinery/poolcontroller.dm (hunk 1 above)
Comment (FalseIncarnate) [inline code/game/machinery/poolcontroller.dm]: You didn't have a comment on that line anyways, but I'll look into adding them

File: code/game/objects/structures/ladders.dm (hunk 2 above)
Comment (FalseIncarnate) [inline code/game/objects/structures/ladders.dm]: They have pre-defined sprites that are static. This was to prevent them from turning back into ladders.

**Comment by Dumbdumn5** [comment]
> Your beachfront may vary. (The water overlay still sucks and is blocky, but that's for another time)
This is randomized? Fucking hell, YES!

**Comment by FalseIncarnate** [comment]
The sand is randomized, the layout is not.

**Comment by Dumbdumn5** [comment]
Awwwww

**Comment by FalseIncarnate** [comment]
Randomizing the layout is a fair bit more tricky, but something i'd like to learn how to do eventually. Randomizing the sand sprites is far easier, and still is a noticeable visual upgrade.

File: code/game/machinery/poolcontroller.dm (hunk 3 above)
Comment (Fox-McCloud) [inline code/game/machinery/poolcontroller.dm]: for consistency, this should be under `handleTemp`, even if it does nothing.

**Pull request merged**